nmtwizard/preprocess/prepoperator.py

```diff
@@ -22,105 +22,147 @@ def get_operator_params(config):
     return config
 
 
+class ProcessType(object):
+      """Type of processing pipeline.
+
+      Possible values are:
+        * ``TRAINING``
+        * ``INFERENCE``
+        * ``POSTPROCESS``
+      """
+
+      TRAINING = 0
+      INFERENCE = 1
+      POSTPROCESS = 2
+
+
 class Pipeline(object):
     """Pipeline for building and applying pre/postprocess operators in order."""
 
-    def __init__(self, config, process_type, preprocess_exit_step):
-        # Current state of preprocessing pipeline.
-        # Passed to and modified by operator initializers if necessary.
-        self.start_state = { "src_tokenizer" : None,
-                             "tgt_tokenizer" : None }
+    def __init__(self, config, preprocess_exit_step):
+        self._build_state = None
+        self._process_type = ProcessType.TRAINING
```

Comment:
There should probably be a `TrainingPipeline`. In that case `Pipeline` should be an abstract class.